src/ipc/StrandCoord.cc

```diff
@@ -37,20 +38,34 @@ void Ipc::StrandCoord::pack(TypedMsgHdr &hdrMsg) const
     hdrMsg.putString(tag);
 }
 
-Ipc::HereIamMessage::HereIamMessage(const StrandCoord &aStrand):
+Ipc::StrandMessage::StrandMessage(const Ipc::MessageType msgType, const StrandCoord &aStrand):
+    messageType(msgType),
     strand(aStrand)
 {
 }
 
-Ipc::HereIamMessage::HereIamMessage(const TypedMsgHdr &hdrMsg)
+Ipc::StrandMessage::StrandMessage(const TypedMsgHdr &hdrMsg):
+    messageType(static_cast<Ipc::MessageType>(hdrMsg.type()))
```

Comment:
Perhaps we need some check for enum ranges here. In this case, we would need to add enumBegin_/enumEnd_ markers to MessageType.

Reply:
I think bad types should be (and arguably already are) handled at message parsing time (i.e. in receive() methods called by Ipc::Port::noteRead()). Outside of that (essentially) parsing/interpreting code, there should be no TypedMsgHdr objects with invalid types roaming around. In fact, I suspect that there should be no TypedMsgHdr objects outside of parsing/packing/unpacking code at all! I hope the existing official and PR code follow that principle already.

If they do, then perhaps splitting TypedMsgHdr::type() into unchecked rawType() and (cheked) type() is not necessary in this PR, but I would not be against such a split if you think it is warranted.

Reply:
> If they do, then perhaps splitting TypedMsgHdr::type() into unchecked rawType() 

Done.